### PR TITLE
[search] Fixed dancing search results.

### PR DESCRIPTION
**Pull request by ygorshenin**

### CODE REVIEW

File: map/framework.cpp [hunk 1]
```diff
@@ -1326,16 +1328,15 @@ void Framework::SetCurrentCountryChangedListener(TCurrentCountryChanged const &
...
 
-void Framework::PokeSearchInViewport()
+void Framework::PokeSearchInViewport(bool forceSearch)
 {
   if (!m_isViewportInitialized || !IsViewportSearchActive())
     return;
 
-  auto & intent = m_searchIntents[static_cast<size_t>(search::Mode::Viewport)];
-  auto & params = intent.m_params;
+  auto params = m_searchIntents[static_cast<size_t>(search::Mode::Viewport)].m_params;
```
Comment (mpimenov) [inline map/framework.cpp]: I guess we need the comment about the intentional copy back.

File: map/framework.hpp [hunk 2]
```diff
@@ -557,7 +557,7 @@ class Framework : public search::ViewportSearchCallback::Delegate,
 
   // When search in viewport is active or delayed, restarts search in
   // viewport.
-  void PokeSearchInViewport();
+  void PokeSearchInViewport(bool forceSearch = true);
```
Comment (mpimenov) [inline map/framework.hpp]: Add a comment to the parameter, please.

File: map/framework.cpp (hunk 1 above)
Comment (ygorshenin) [reply map/framework.cpp]: Done

File: map/framework.hpp (hunk 2 above)
Comment (ygorshenin) [reply map/framework.hpp]: Done

**Pull request merged**